src/main/java/io/vertx/core/net/impl/clientconnection/Pool.java

```diff
@@ -310,6 +291,26 @@ private Runnable nextTask() {
         };
       }
     }
+    if (waitersQueue.size() > 0) {
```

Comment:
@vietj we will have to check if connections are valid and expire them before assigning the connection.